cluster/cluster.go

```diff
@@ -54,6 +55,7 @@ type Cluster struct {
 const (
 	X509AuthenticationProvider = "x509"
 	StateConfigMapName         = "cluster-state"
+	FullStateConfigMapName     = "full-cluster-state"
```

Comment:
what is the diff between StateConfigMapName (cluster-state) and FullStateConfigMapName (full-cluster-state)?

Reply:
StateConfigMapName is the name we used with "legacy" cluster state. FullStateConfigMapName is the name of the new Desired/Current state we save on kubernetes. We renamed it because we actually delete StateConfigMapName when we handle the legacy cluster upgrade. 

StateConfigMapName will be removed in the future.